static/app/views/settings/organizationDeveloperSettings/permissionsObserver.tsx

```diff
@@ -55,14 +61,48 @@ export default class PermissionsObserver extends Component<Props, State> {
 
   onPermissionChange = (permissions: Permissions) => {
     this.setState({permissions});
+    const new_permissions = toResourcePermissions(this.props.scopes);
+
+    let elevating = false;
+    Object.keys(permissions).some((resource_name: string) => {
+      if (
+        comparePermissionLevels(
+          permissions[resource_name],
+          new_permissions[resource_name]
+        ) > 0
+      ) {
+        elevating = true;
+        return true;
+      }
+      return false;
+    });
+
+    this.setState({elevating});
   };
 
   onEventChange = (events: WebhookEvent[]) => {
     this.setState({events});
   };
 
+  renderCallout() {
+    const {elevating} = this.state;
+
+    if (elevating === true) {
+      return (
+        <Alert type="warning" showIcon>
+          {t(
+            'You are going to increase privileges for this integration. Organization members who already had access to the Client Secret may gain extra permissions due to this change. If this is not what you are expecting, consider re-creating an integration.'
```

Comment:
This sounds slightly better imo
```suggestion
            'You are going to increase privileges for this integration. Organization members who already had access to the Client Secret may gain extra permissions due to this change. If this is not what you are expecting, consider re-creating the integration.'
```